ocaml/xapi/export_raw_vdi.ml

```diff
@@ -51,14 +45,29 @@ let localhost_handler rpc session_id vdi (req: Http.Request.t) (s: Unix.file_des
              raise (Api_errors.Server_error (Api_errors.vdi_io_error, ["Device I/O errors"])) in
          begin
            try
-             Sm_fs_ops.with_block_attached_device __context rpc session_id vdi `RO
-               (fun path ->
-                  match Importexport.base_vdi_of_req ~__context req with
-                  | Some base_vdi ->
-                    Sm_fs_ops.with_block_attached_device __context rpc session_id base_vdi `RO
-                      (fun base_path -> copy (Some base_path) path)
-                  | None -> copy None path
-               )
+             let headers = Http.http_200_ok ~keep_alive:false () @ [
+                 Http.Hdr.task_id ^ ":" ^ (Ref.string_of task_id);
+                 Http.Hdr.content_type ^ ":" ^ content_type;
+                 Http.Hdr.content_disposition ^ ": attachment; filename=\"" ^ filename ^ "\""
+               ] in
+             Http_svr.headers s headers;
+             match format with
+             | Raw | Vhd ->
+               Sm_fs_ops.with_block_attached_device __context rpc session_id vdi `RO
+                 (fun path ->
+                    match Importexport.base_vdi_of_req ~__context req with
+                    | Some base_vdi ->
+                      Sm_fs_ops.with_block_attached_device __context rpc session_id base_vdi `RO
+                        (fun base_path -> copy (Some base_path) path)
+                    | None -> copy None path
+                 )
+             | Tar ->
+               (* We need to keep refreshing the session to avoid session timeout *)
+               let refresh_session = Xapi_session.consider_touching_session rpc session_id in
+               (* The third element of the tuple, the size, is currently ignored by the function *)
+               let size = Client.Client.VDI.get_virtual_size ~rpc ~session_id ~self:vdi in
+               Stream_vdi.send_all refresh_session s __context rpc session_id [(Xapi_globs.vdi_tar_export_dir, vdi, size)];
+               Tar_unix.write_end s;
```

Comment:
Where did we write the beginning?

Reply:
I think we don't need it, I followed the example of `export.ml` where we just call this at the end.
In `send_all`, we write the header for each file.
I'v enabled import-export quicktests in tar format with randomized data, and they seem to work fine.
I also managed to untar the export, it only complained about invalid dates.